x/ecocredit/client/testsuite/tx.go

```diff
@@ -1445,165 +966,132 @@ func (s *IntegrationTestSuite) TestTxSell() {
+					fmt.Sprintf("[{batch_denom: \"%s\", quantity: \"5\", ask_price: \"100uregen\", disable_auto_retire: false, expiration: \"2024-01-01\"}]", batchDenom),
 					makeFlagFrom(val0.Address.String()),
 				},
 				s.commonTxFlags()...,
 			),
-			sellOrderId: "5",
-			expErr:      false,
-			expOrder: &ecocredit.SellOrder{
-				OrderId:           5,
-				Owner:             val0.Address.String(),
-				BatchDenom:        batchDenom,
+			expErr: false,
+			expOrder: &marketplace.SellOrder{
+				Seller:            val0.Address,
 				Quantity:          "5",
-				AskPrice:          &sdk.Coin{Denom: "regen", Amount: sdk.NewInt(100)},
+				AskPrice:          "100",
 				DisableAutoRetire: false,
-				Expiration:        &expiration,
+				Expiration:        types.ProtobufToGogoTimestamp(timestamppb.New(expiration)),
 			},
 		},
 	}
 
 	for _, tc := range testCases {
 		s.Run(tc.name, func() {
 			cmd := marketplaceclient.TxSellCmd()
-			_, err := cli.ExecTestCLICmd(clientCtx, cmd, tc.args)
+			out, err := cli.ExecTestCLICmd(clientCtx, cmd, tc.args)
 			if tc.expErr {
 				s.Require().Error(err)
 				s.Require().Contains(err.Error(), tc.expErrMsg)
 			} else {
 				s.Require().NoError(err)
+				var res sdk.TxResponse
+				s.Require().NoError(clientCtx.Codec.UnmarshalJSON(out.Bytes(), &res))
+				s.Require().True(len(res.Logs) > 0)
 
-				// query sell order
-				query := marketplaceclient.QuerySellOrderCmd()
-				out, err := cli.ExecTestCLICmd(clientCtx, query, []string{
-					tc.sellOrderId,
-					flagOutputJSON,
-				})
-				s.Require().NoError(err, out.String())
-
-				// unmarshal query response
-				var res ecocredit.QuerySellOrderResponse
-				err = clientCtx.Codec.UnmarshalJSON(out.Bytes(), &res)
-				s.Require().NoError(err)
-
-				// verify expected order
-				s.Require().Equal(tc.expOrder, res.SellOrder)
+				found := false
+				for _, e := range res.Logs[0].Events {
+					if e.Type == proto.MessageName(&marketplace.EventSell{}) {
+						for _, attr := range e.Attributes {
+							if attr.Key == "order_id" {
+								found = true
+								orderIdStr := strings.Trim(attr.Value, "\"")
+								_, err := strconv.ParseUint(orderIdStr, 10, 64)
+								s.Require().NoError(err)
+								queryCmd := marketplaceclient.QuerySellOrderCmd()
+								queryArgs := []string{orderIdStr, flagOutputJSON}
+								queryOut, err := cli.ExecTestCLICmd(clientCtx, queryCmd, queryArgs)
+								s.Require().NoError(err, queryOut.String())
+								var queryRes marketplace.QuerySellOrderResponse
+								s.Require().NoError(clientCtx.Codec.UnmarshalJSON(queryOut.Bytes(), &queryRes))
+								s.Require().Equal(queryRes.SellOrder.Quantity, tc.expOrder.Quantity)
+								s.Require().Equal(tc.expOrder.DisableAutoRetire, queryRes.SellOrder.DisableAutoRetire)
+								s.Require().True(tc.expOrder.Expiration.Equal(queryRes.SellOrder.Expiration))
+								break
+							}
+							if found {
+								break
+							}
+						}
+					}
+					if found {
+						break
+					}
```

Comment:
If we are removing test cases in client tests that are covered in unit tests, I don't think we need to be checking the event logs here. We also return `sell_order_ids` in `MsgSellResponse`.

Reply:
it's not just checking the event logs, it's also checking the state transition happened correctly, by getting the order Id's and then querying for the sell orders. 

the output of a tx CLI command is a TxResponse, so i cannot use MsgSellResponse here

Reply:
Hmmmm... I think this is something that should already be covered in unit tests. I think we should simply be checking if the command was executed without errors when testing the commands. Maybe this should also be addressed separately as part of #1041.